File: two-pointers/01-valid-palindrome/MainTest.java
public class MainTest {
    static int passed = 0;
    static int failed = 0;

    public static void main(String[] args) {
        check("example 1", () -> {
            boolean result = new Solution().isPalindrome("Was it a car or a cat I saw?");
            assertEqual(true, result);
        });

        check("example 2", () -> {
            boolean result = new Solution().isPalindrome("tab a cat");
            assertEqual(false, result);
        });

        System.out.println(passed + " passed, " + failed + " failed");
        if (failed > 0) {
            System.exit(1);
        }
    }

    interface TestCase {
        void run() throws Exception;
    }

    static void check(String name, TestCase test) {
        try {
            test.run();
            passed++;
            System.out.println("PASS: " + name);
        } catch (Throwable e) {
            failed++;
            System.out.println("FAIL: " + name + " -> " + e.getMessage());
        }
    }

    static void assertEqual(Object expected, Object actual) {
        if (!java.util.Objects.equals(expected, actual)) {
            throw new AssertionError("expected " + expected + " but got " + actual);
        }
    }
}

File: arrays-hashing/07-product-of-array-except-self/main_test.py
import unittest

from main import Solution


class TestProductExceptSelf(unittest.TestCase):
    def test_example_1(self):
        sol = Solution()
        self.assertEqual(sol.productExceptSelf([1, 2, 4, 6]), [48, 24, 12, 8])

    def test_example_2(self):
        sol = Solution()
        self.assertEqual(sol.productExceptSelf([-1, 0, 1, 2, 3]), [0, -6, 0, 0, 0])


if __name__ == "__main__":
    unittest.main()
